OMCompiler/Compiler/NFFrontEnd/NFExpandableConnectors.mo

```diff
@@ -285,18 +287,38 @@ protected
   ComponentRef virtual_cref, normal_cref;
   Type ty;
   InstNode node;
+  list<Subscript> subs;
+  Subscript s1;
+  Expression expr;
+  Integer arrIndex;
 algorithm
   virtual_cref := virtualConnector.name;
   normal_cref := normalConnector.name;
   ty := normalConnector.ty;
 
+  subs := ComponentRef.getSubscripts(virtual_cref);
+  if (listLength(subs) > 0) then
+    if not (listLength(subs) == 1) then
+      print("to many subscripts in expandable connector\n");
+    end if;
+    s1 := List.first(subs);
+    expr := Subscript.toExp(s1);
+    expr := Ceval.evalExp(expr);
+    if not Type.isInteger(Expression.typeOf(expr)) then
+      print("unknown type in expandable connector\n");
+    end if;
```

Comment:
I don't think this is correct, a subscript can also be an enumeration or Boolean. Since the type should already have been checked during type checking it's probably not necessary to check it here anyway.